### PR TITLE
VB-3742 Block visit page - call API to add new blocked date

### PR DESCRIPTION
**Pull request by tpmcgowan**
'Wire-up' the block a date confirmation page so it calls the API to actually block the date.

N.B. Various `TODO`s present that will be sorted in upcoming PRs...

### CODE REVIEW

File: assets/sass/local.scss [hunk 1]
```diff
@@ -226,11 +226,6 @@
   }
 }
 
-#datepicker-date {
```
Comment (tpmcgowan) [inline assets/sass/local.scss:229]: Unrelated - spotted some left-over CSS from our previous date picker implementation.

File: server/views/pages/visits/summary.njk [hunk 2]
```diff
@@ -54,7 +54,6 @@
                     text: "For example, 31/3/2022"
                   },
                   errorMessage: errors | findError('date'),
-                  classes: 'hmpps-datepicker--fixed-width',
```
Comment (tpmcgowan) [inline server/views/pages/visits/summary.njk:57]: Unrelated - stray class from previous date picker implementation

File: server/data/orchestrationApiClient.test.ts [hunk 3]
```diff
@@ -416,16 +417,35 @@ describe('orchestrationApiClient', () => {
...
+        .matchHeader('authorization', `Bearer ${token}`)
+        .reply(200)
+
+      await orchestrationApiClient.blockVisitDate(prisonId, date, user)
+
+      expect(fakeOrchestrationApi.isDone()).toBe(true)
+    })
+  })
+
   describe('getFutureExcludeDates', () => {
```
[the remark this answers is not in the document]
Comment (hutcheonb-moj) [reply server/data/orchestrationApiClient.test.ts:439]: I followed the naming directly from the orchestration, but yeah the mix of naming will be a problem - we just have to decide at what level to change it over

File: server/data/orchestrationApiClient.ts [hunk 4]
```diff
@@ -249,9 +256,9 @@ export default class OrchestrationApiClient {
     })
   }
 
-  async getPrison(prisonCode: string): Promise<PrisonDto> {
+  async getPrison(prisonId: string): Promise<PrisonDto> {
     return this.restClient.get({
-      path: `/config/prisons/prison/${prisonCode}`,
+      path: `/config/prisons/prison/${prisonId}`,
```
Comment (hutcheonb-moj) [inline server/data/orchestrationApiClient.ts:261]: The prisonId / prisonCode bring immense frustration from my point of view, probably better if we do decide to always use the same one ignoring the requirement of the orchestration

**Review by hutcheonb-moj — APPROVED** [review]
LGTM

File: server/data/orchestrationApiClient.ts (hunk 4 above)
Comment (tpmcgowan) [reply server/data/orchestrationApiClient.ts:261]: It's pretty consistent in this repo which is why I changed this. Admin UI is all over the place with code vs id 😂

**Pull request merged**